pkg/api/validate.go

```diff
@@ -148,6 +148,10 @@ func validateUpdateContainerService(cs, oldCs *OpenShiftManagedCluster, external
 		}
 		old.Properties.AgentPoolProfiles[i].Count = new.Count
 	}
+	// QUESTION: Unclear whether this is correct. We can track the state
+	// separately inside the fake rp but need to ensure first this is
+	// something that MSFT does not need.
+	old.Properties.ProvisioningState = cs.Properties.ProvisioningState
```

Comment:
Yes, this is probably wrong and users should not be able to change this field. Will remove this and make the fake RP set it after validation.